### PR TITLE
v2: add support for loading root ca certificate file

### PR DESCRIPTION
**Pull request by sarge**
Adds support for loading a custom root ca file via the caddyfile.

```
tls [off|<cert> <key>] {
	load <folders...>
	protocols <min> [<max>]
	ciphers <list...>
	curves <list...>
	alpn <list...>
        ca <url>
        ca_root <file>
}
```

### CODE REVIEW

**Comment by francislavoie** [comment]
https://github.com/caddyserver/caddy/pull/2936 Seems related.

**Comment by zaquestion** [comment]
Interesting, to see that these certs are loaded from files using the proposed approach in https://github.com/caddyserver/caddy/pull/2936

Ref: https://github.com/caddyserver/caddy/blob/5ab17a3a371c1f84ef5a649d9310971eb7959271/modules/caddytls/acmemanager.go#L130-L142

**Comment by mholt** [comment]
Oh, just resolve the merge conflict first, sorry :)

File: caddyconfig/httpcaddyfile/options.go [hunk 1]
```diff
@@ -174,6 +174,18 @@ func parseOptACME(d *caddyfile.Dispenser) (string, error) {
...
+	if !d.Next() {
+		return "", d.ArgErr()
+	}
+	val := d.Val()
+	if d.Next() {
+		return "", d.ArgErr()
+	}
+	return val, nil
+}
+
```
Comment (mholt) [inline caddyconfig/httpcaddyfile/options.go]: Since this is the exact same as `parseOptACME`, just use `parseOptACME` instead.

**Comment by sarge** [comment]
Closing in favour of https://github.com/caddyserver/caddy/pull/3040

**Pull request closed without merging**